### ISSUE TITLE
Very slow performance

### ISSUE THREAD

**Issue by alexandrehtrb** [kuiperzone/AvantGarde#36]
Hello,

I tried using AvantGarde for my project and the UI froze a lot... 

My setup:

Windows 11 x64
Intel Core i5
16GB RAM

**Comment by kuiperzone** [comment]
I'm not seeing this, but I guess it might slow down with large projects. How big is your project/solution. Does it have many modules etc?

**Comment by alexandrehtrb** [comment]
> I'm not seeing this, but I guess it might slow down with large projects. How big is your project/solution. Does it have many modules etc?

I think it is not big

https://github.com/alexandrehtrb/Pororoca

**Comment by kuiperzone** [comment]
I tried your project. The first thing I noticed was the high CPU usage when viewing your `HttpRequestView.xaml` file (see screenshot). The thing is that the backend Preview Host is actually running your code and it seems most of the high CPU usage in this case is coming from your code, but only with this particular form. Your other forms seem fine.

However, the CPU usage with Avant Garde viewing a project is a little higher than I would like generally. I'm not sure I can do much about this as it must watch your files for changes. I can, however, make the main thread more responsive in the next release.
